apps/daimo-mobile/src/view/screen/OnboardingScreen.tsx

```diff
@@ -298,16 +272,16 @@ function TryEnclave({
       onNext();
     } catch (e: any) {
       console.error(e);
-      if (!tryWeaker) {
-        console.error(
-          `[ONBOARDING] enclave signature trial error, asking to downgrade to weaker keys`
-        );
-        setAskWeak(true);
+      if (e instanceof NamedError && e.name === "ExpoEnclaveSign") {
+        setError(e.message);
       } else {
-        const err = `[ONBOARDING] Failed with weaker keys too, giving up`;
-        setError(err);
-        console.error(err);
+        setError("Unknown error");
       }
+
+      // Assume user doesn't have proper auth set up.
+      // TODO: In future, catch different errors differently and
+      // show a retry button in case user intentionally cancelled auth.
+      setAskToSetPin(true);
```

Comment:
we can show a catch-all like `Auth failed. PIN not set?`

Reply:
we display the error from expo enclave module in red and we do have a try again button now so that comment is slightly outdated (will fix). Will include your other copy suggestion and then I think we should be good to go.